### PR TITLE
Distributions with deleted nonsence values and with cuts based on phy…

### PR DESCRIPTION
**Pull request by conformist89**
…sical processes or detector geometry

### CODE REVIEW

**Comment by viktorklochkov** [comment]
Hi @conformist89! Usually, .pdf and .docx files are not stored in git, maybe you could remove them and leave only code.

**Comment by viktorklochkov** [comment]
Also, is there a way to move code to .py file? In .ipynb there are also output cells

**Comment by shahidzk1** [comment]
> Also, is there a way to move code to .py file? In .ipynb there are also output cells

We can store an .ipynb file without the output, save just the raw file and upload it. I am working on creating a .py file

**Review by viktorklochkov — APPROVED** [review]

**Comment by viktorklochkov** [comment]
@shahidzk1, please approve the changes if they are fine and we can merge it

**Comment by shahidzk1** [comment]
@conformist89 the concurrent package was used in the tree_importer package to import in stuff. if you aren't using it in the rest of the code then try to remove it. i have suggested this change. Also it creates a dataframe out of a tree so don't create df again in line 30. This will help you save some memory. Try to make a function named quality cuts and use that instead of doing everyting for signal and background separately. Like our data clean function.

File: quality_cuts/new_var_dist.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import gc
+
+"""
+ All the nonesences ( $\chi$2 < 0), inf and nan were deleted. Also applied
+ quality cuts based on detector geometry. Full description could be found in
+ https://docs.google.com/document/d/11f0ZKPW8ftTVhTxeWiog1g6qdsGgN1mlIE3vd5FHLbc/edit?usp=sharing
+
+"""
+
+from concurrent.futures import ThreadPoolExecutor
```
Comment (shahidzk1) [inline quality_cuts/new_var_dist.py:19]: remove these

File: quality_cuts/new_var_dist.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+executor = ThreadPoolExecutor(8)
+
+
+
+file = tree_importer('/home/olha/CBM/dataset/10k_events_PFSimplePlainTree.root',
```
Comment (shahidzk1) [inline quality_cuts/new_var_dist.py]: change file to df_original, it is already a dataframe

File: quality_cuts/new_var_dist.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+"""
+ All the nonesences ( $\chi$2 < 0), inf and nan were deleted. Also applied
+ quality cuts based on detector geometry. Full description could be found in
+ https://docs.google.com/document/d/11f0ZKPW8ftTVhTxeWiog1g6qdsGgN1mlIE3vd5FHLbc/edit?usp=sharing
+
+"""
+
+from concurrent.futures import ThreadPoolExecutor
+executor = ThreadPoolExecutor(8)
```
Comment (shahidzk1) [inline quality_cuts/new_var_dist.py]: remove these if you don't use them, they are already included in the tree_importer file

File: quality_cuts/new_var_dist.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+df_original.columns = new_labels
+
+
+sgnal = df_original[df_original['issignal']==1]
+
+
+bg = df_original[df_original['issignal']==0]
+
+
```
Comment (shahidzk1) [inline quality_cuts/new_var_dist.py:48]: make a function and save it in a separate python file as quality_cuts. The other cuts will be named as pre-selection cuts

**Review by shahidzk1 — CHANGES_REQUESTED** [review]

**Review by shahidzk1 — APPROVED** [review]
Olga try to write also in English what the code is doing. In the future you will remember why did you write this code the way it is and also others can read it as well. I also don't elaborate my code but that is generally a bad practice.

**Comment by shahidzk1** [comment]
@viktorklochkov this work should be saved in data cleaning, correlations one is a different work. We should create another repository for it. One for correlations and one for data cleaning and pre-processing.

**Comment by conformist89** [comment]
> .

I added comments

**Review by shahidzk1 — APPROVED** [review]

**Review by shahidzk1 — APPROVED** [review]

**Pull request merged**